state/factory/sovereignAccountCreator.go

```diff
@@ -0,0 +1,67 @@
+package factory
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/hashing"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/state/accounts"
+	"github.com/multiversx/mx-chain-go/state/parsers"
+	"github.com/multiversx/mx-chain-go/state/trackableDataTrie"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+// sovereignAccountCreator has method to create a new account
+type sovereignAccountCreator struct {
+	hasher              hashing.Hasher
+	marshaller          marshal.Marshalizer
+	esdtAsBalance       state.ESDTAsBalanceHandler
+	enableEpochsHandler common.EnableEpochsHandler
+}
+
+// NewSovereignAccountCreator creates a new instance of AccountCreator
+func NewSovereignAccountCreator(args ArgsAccountCreator) (state.AccountFactory, error) {
+	if check.IfNil(args.Hasher) {
+		return nil, errors.ErrNilHasher
+	}
+	if check.IfNil(args.Marshaller) {
+		return nil, errors.ErrNilMarshalizer
+	}
+	if check.IfNil(args.EnableEpochsHandler) {
+		return nil, errors.ErrNilEnableEpochsHandler
+	}
+
+	esdtAsBalance, err := accounts.NewESDTAsBalance("", args.Marshaller)
```

Comment:
the token ID will need to be taken from the configs

Reply:
exactly.

Reply:
Yes, it will be done in a future pr